org-code-javabuilder/lib/src/main/java/org/code/javabuilder/AWSInputAdapter.java

```diff
@@ -0,0 +1,47 @@
+package org.code.javabuilder;
+
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+
+/** Accesses Amazon SQS to get user input for the currently running program. */
+public class AWSInputAdapter implements InputAdapter {
+  private final AmazonSQS sqsClient;
+  private final String queueUrl;
+  private final Queue<String> messages;
+
+  public AWSInputAdapter(AmazonSQS sqsClient, String queueUrl) {
+    this.sqsClient = sqsClient;
+    this.queueUrl = queueUrl;
+    this.messages = new LinkedList<>();
+  }
+
+  /**
+   * Attempts to access the first element in the fifo queue. If none exist, queries available input
+   * from Amazon SQS and stores each input in the `messages` fifo queue. Messages are deleted after
+   * they are retrieved. This is a blocking call.
+   *
+   * @return the first message in the fifo queue.
+   */
+  public String getNextMessage() {
+    ReceiveMessageRequest request = new ReceiveMessageRequest();
+    request.setQueueUrl(queueUrl);
+    request.setWaitTimeSeconds(20);
```

Comment:
for my information: how does this wait time work? Does it wait up to 20 seconds on each call to getMessages()? Something else? (may be worth adding a comment)

Reply:
Yes, that's what it does - it could return much faster, but 20s is the max. I'll add a comment.